modules/packages/Sort.chpl

```diff
@@ -18,53 +18,321 @@
-   default. If `true`, then the sort function will order in reverse.
+
+This module supports a variety of standard sorting routines on 1D arrays.
+
+The current interface is minimal and should be expected to
+grow and evolve over time.
+
+Comparators
+-----------
+
+Comparators allow sorting data by a mechanism other than the
+default comparison of array elements. To use a comparator, define a
+record with either a ``key(a)`` or ``compare(a, b)`` method, and pass an
+instance of that record to the sort function (examples shown below).
+
+If both methods are implemented on the record passed as the comparator, the
+``key(a)`` method will take priority over the ``compare(a, b)`` method.
+
+Nearly all sort routines support comparators, which is denoted by their
+function signature.
+
+Key Comparator
+~~~~~~~~~~~~~~
+
+The ``key(a)`` method accepts 1 argument, which will be an element from the
+array being sorted. The value returned should support numeric operations.
+
+The default key method would look like this:
+
+.. code-block:: chapel
+
+  proc DefaultComparator.key(a) {
+    return a;
+  }
+
+
+As an example, if the user wants to sort an array by the absolute values of its
+elements, the user can define a comparator with a key method as follows:
+
+.. code-block:: chapel
+
+  var Array = [-1, -4, 2, 3];
+
+  // Empty record serves as comparator
+  record Comparator { }
+
+  // key method maps an element to the value to be used for comparison
+  proc Comparator.key(a) { return abs(a); }
+
+  var absComparator: Comparator;
+
+  QuickSort(Array, comparator=absComparator);
+
+  // This will output: -1, 2, 3, -4
+  writeln(Array);
+
+
+Compare Comparator
+~~~~~~~~~~~~~~~~~~
+
+The ``compare(a, b)`` method accepts 2 arguments, which will be 2 elements from
+the array being sorted. The return value should be an integer indicating how a
+and b compare to each other, as shown below:
+
+  * > 0 if ``a > b``
+  * 0 if ``a == b``
+  * < 0 if ``a < b``
+
+The default compare method for a numeric signed type would look like this:
+
+.. code-block:: chapel
+
+    proc DefaultComparator.compare(a, b) {
+      return a - b;
+    }
+
+
+The absolute value comparison example from above can alternatively be
+implemented with a compare method:
+
+.. code-block:: chapel
+
+  var Array = [-1, -4, 2, 3];
+
+  // Empty record serves as comparator
+  record Comparator { }
+
+  // compare method defines how 2 elements are compared
+  proc Comparator.compare(a, b) {
+    return abs(a) - abs(b);
+  }
+
+  var absComparator: Comparator;
+
+  QuickSort(Array, comparator=absComparator);
+
+  // This will output: -1, 2, 3, -4
+  writeln(Array);
+
+Reverse Comparator
+~~~~~~~~~~~~~~~~~~
+
+Sort functions in Chapel do not have a ``reverse`` argument. Instead, reverse
+sorting is handled through the comparator interface.
+
+A module-defined :const:`reversecomparator` can be passed to a sort function to
+reverse the default sorting order.
```

Comment:
I would have expected it to be named reverseComparator. Is there a reason you opted for the all-lower-case version?